app/microsoft/base.py

```diff
@@ -48,11 +35,13 @@ def handle(self, resp):
         try:
```

Comment:
Refactored this in two ways:
(1) no longer using pprint module functions, found that messages being logged were actually neater without it.
(2) if the request failed because item wasn't found, will log an error and return None (this is informative and can be used in our logic), otherwise will log error and raise HTTPError ...